pkg/cloud/talos/provisioners/azure/azure.go

```diff
@@ -0,0 +1,416 @@
+func (azure *Az) Update(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+	return nil
+}
+
+// Delete deletes a Azure instance.
+func (azure *Az) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	azureConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), azureConfig)
+
+	vmClient, err := vmclient()
+	if err != nil {
+		return err
+	}
+
+	// Submit the delete request
+	vmfuture, err := vmClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name)
+	if err != nil {
+		return err
+	}
+
+	err = vmfuture.WaitForCompletionRef(ctx, vmClient.Client)
+	if err != nil {
+		return err
+	}
+
+	// Cleanup os disk
+	disksClient, err := disksclient()
+	if err != nil {
+		return err
+	}
+	_, err = disksClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name+"-os-disk")
+	if err != nil {
+		return err
+	}
+
+	// Cleanup nic
+	nicClient, err := nicclient()
+	if err != nil {
+		return err
+	}
+	_, err = nicClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name+"-nic")
+	if err != nil {
+		return err
+	}
+
+	log.Println("[Azure] Instance deleted: " + machine.ObjectMeta.Name)
+
+	return nil
+}
+
+// Exists returns whether or not an instance is present in Azure.
+func (azure *Az) Exists(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) (bool, error) {
+
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return false, err
+	}
+
+	azureConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), azureConfig)
+
+	vmClient, err := vmclient()
+	if err != nil {
+		return true, err
+	}
+
+	// If there's an error from retrieving the VM by name, assume it doesn't exist
+	_, err = getVMByName(ctx, vmClient, azureConfig.ResourceGroup, machine.ObjectMeta.Name)
+	if err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// getSubnetByName finds a given subnet (required for input along with network)
+func getSubnetByName(ctx context.Context, azureConfig *ClusterInfo) (*network.Subnet, error) {
+
+	client, err := subnetclient()
+	if err != nil {
+		return nil, err
+	}
+
+	subnet, err := client.Get(ctx, azureConfig.ResourceGroup, azureConfig.Instances.Network, azureConfig.Instances.Subnet, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return &subnet, nil
+}
+
+// getPublicIPbyIP finds the public IP object from a list of all IP objects
+func getPublicIPbyIP(ctx context.Context, ipAddress string) (*network.PublicIPAddress, error) {
+
+	client, err := ipclient()
+	if err != nil {
+		return nil, err
+	}
+
+	//Dump all public IPs created and iterate to find our desired IP
+	list, err := client.ListAllComplete(ctx)
```

Comment:
```suggestion
	list, err := client.ListComplete(ctx)
```
If you're not allowing multi-resourcegroup for now, `ListCOmplete` will be much more efficient.